controller/eks-cluster-config-handler.go

```diff
@@ -444,7 +449,16 @@ func (h *Handler) create(config *eksv1.EKSClusterConfig, sess *session.Session,
 
 	displayName := config.Spec.DisplayName
 
-	var err error
+	listOutput, err := eksService.ListClusters(&eks.ListClustersInput{})
+	if err != nil {
+		return config, fmt.Errorf("error listing clusters: %v", err)
+	}
+	for _, cluster := range listOutput.Clusters {
+		if aws.StringValue(cluster) == displayName {
+			return config, fmt.Errorf("cannot create cluster [%s] because a cluster in EKS exists with the same name", displayName)
+		}
+	}
+
```

Comment:
Don't we need to protect from multiple create calls that happen parallely? Its possible to use the same cluster name if the create happens almost at the same time. Any lock needed around this?

Reply:
@rmweir Is that something we need to worry about for this? I know there is some login in rancher to disallow creating a cluster with the same name as one that already exists in rancher, but not sure if that is concurrent-safe.

Reply:
There is a validator which seeks to prevent this. Since it uses the cache it is possible but unlikely this would happen unless someone was programmatically making clusters that reference the same EKS cluster. Something that could happen is an object being reenqueued due to an update below and going through this block again before being set to "active". However, I think this is also unlikely and not sure an UpdateStatus call causes a reenqueue in rancher; just something to keep an eye on.

Reply:
We could just use a local mutex to lock the creation - this would save from concurrent calls made to same server instance - [For an HA setup, concurrent calls can still happen if they landup on different instances.]

Reply:
@prachidamle Idk how this would be more prevalent on an HA setup since only one instance is deployed and the deployment code is launch from the management context. I don't think we should be adding mutex's premptively since we don't have evidence this would occur.

Reply:
Oh good if there will be only single instance in HA then a mutex would definitely work for parallel requests coming in. We should test this by making parallel create calls and using same clustername.

Reply:
We have used such a mutex, in Rancher's cluster store earlier

Reply:
I'm not sure if `eks-operator` is the right place for the mutex because the request isn't going to `eks-operator`. The request goes to `rancher`, it creates the `eksoperatorconfig` resource, then `eks-operator` gets alerted to the existence of the resource.

I'm still learning about the `rancher` API, but perhaps the very mutex your linked to will handle the case you are referring to.

Reply:
Ok, is it possible to test it out? If it's the Rancher API call, we can call it parallely to see if same name can get used by eks?

Reply:
Yes. I tested that the mutex in the linked code protects against multiple clusters resources from being created with the same display name.